code/game/machinery/vending.dm

```diff
@@ -566,6 +569,28 @@
 		credit_purchase(customer_account.owner_name)
 		return 1
 
+// Pay with an alternative currency
+/obj/machinery/vending/proc/pay_with_item(obj/item/I, mob/user)
+	var/should_qdel = TRUE
+	var/amount_to_spend = currently_vending.price
+
+	if(istype(I, /obj/item/stack))
+		var/obj/item/stack/S = I
+		if(S.amount >= amount_to_spend)
+			S.use(amount_to_spend)
+			should_qdel = FALSE		// Don't qdel a stack with remaining charges
+		else
+			to_chat(user, SPAN_WARNING("\icon[I] That is not enough money."))
+			return FALSE
```

Comment:
```suggestion
	if(istype(I, /obj/item/stack))
		var/obj/item/stack/S = I
		if(S.amount >= amount_to_spend)
			S.use(amount_to_spend)
			if(S.amount)
				should_qdel = FALSE		// Don't qdel a stack with remaining charges
		else
			to_chat(user, SPAN_WARNING("\icon[I] That is not enough money."))
			return FALSE
	else
			return FALSE
```
You just still delete the stack if the person has exactly enough money since there is 0 remaining.
Also unction is not supposed to be called with a non-stack right? Better be safe by returning FALSE otherwise the message will pop and item will be deleted.